File: LMIS/LMIS.Web/Modules/BookManage/Book/BookColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace LMIS.BookManage.Columns;

[ColumnsScript("BookManage.Book")]
[BasedOnRow(typeof(BookRow), CheckNames = true)]
public class BookColumns
{
    [EditLink, AlignRight]
    public long BookId { get; set; }
    [EditLink]
    public string BookName { get; set; }
    public string CN { get; set; }
    public string ISSN { get; set; }
    public string ISBN { get; set; }
    public string AuthorName { get; set; }
    public string PublisherName { get; set; }
    [Width(100)]
    [DateTimeFormatter(DisplayFormat = "yyyy/MM/dd")]
    public DateTime PublisheDate { get; set; }
    public int Page { get; set; }
    public decimal Price { get; set; }
    public long Inventory { get; set; }
    public long BorrowableInventory { get; set; }
    public short BookStatus { get; set; }
    [Width(150)]
    [DateTimeFormatter(DisplayFormat = "yyyy/MM/dd HH:mm:ss")]
    public DateTime UpdateTime { get; set; }
}

File: LMIS/LMIS.Web/Modules/ServerTypes/BookManage/BookColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { BookStatusEnum } from "../Modules/BookManage.Book.BookStatusEnum";
import { BookRow } from "./BookRow";

export interface BookColumns {
    BookId: Column<BookRow>;
    BookName: Column<BookRow>;
    CN: Column<BookRow>;
    ISSN: Column<BookRow>;
    ISBN: Column<BookRow>;
    AuthorName: Column<BookRow>;
    PublisherName: Column<BookRow>;
    PublisheDate: Column<BookRow>;
    Page: Column<BookRow>;
    Price: Column<BookRow>;
    Inventory: Column<BookRow>;
    BorrowableInventory: Column<BookRow>;
    BookStatus: Column<BookRow>;
    UpdateTime: Column<BookRow>;
}

export class BookColumns extends ColumnsBase<BookRow> {
    static readonly columnsKey = 'BookManage.Book';
    static readonly Fields = fieldsProxy<BookColumns>();
}

[BookStatusEnum]; // referenced types